fingertip/util/reflink.py

```diff
@@ -30,3 +38,86 @@ def is_supported(dirpath):
     if r.returncode and not sure_not:
         log.error('reflink support detection inconclusive, cache dir problems')
     return r.returncode == 0
+
+
+def create_supported_fs(backing_file, size):
+    subprocess.run(['fallocate', '-l', size, backing_file], check=True)
+    subprocess.run(['mkfs.xfs', '-m', 'reflink=1', backing_file],
+                   check=True)
+
+
+def mount_supported_fs(backing_file, tgt):
+    # use warning to avoid rewriting the prompt with sudo prompt
+    log.warning('mounting a reflink-supported filesystem for image storage...')
+    tgt_uid, tgt_gid = os.stat(tgt).st_uid, os.stat(tgt).st_gid
+    subprocess.run(['sudo', 'mount', '-o', 'loop', backing_file, tgt],
+                   check=True)
+    mount_uid, mount_gid = os.stat(tgt).st_uid, os.stat(tgt).st_gid
+    if (tgt_uid, tgt_gid) != (mount_uid, mount_gid):
+        log.debug(f'fixing owner:group ({tgt_uid}:{tgt_gid})')
+        subprocess.run(['sudo', 'chown', f'{tgt_uid}:{tgt_gid}', tgt],
+                       check=True)
+
+
+def storage_setup_wizard():
+    assert SETUP in ('auto', 'suggest', 'never')
+    if SETUP == 'never':
+        return
+    size = SIZE
+    machinesdir = MACHINES
+    if not os.path.exists(machinesdir):
+        os.mkdir(machinesdir)
+    backing_file = os.path.join(machinesdir, 'for-machines.xfs')
```

Comment:
Is it intentional that the backing file lies below the directory it's mounted into? It works nonetheless, but it might be confusing, I guess.

Reply:
That is how you showed me to do that on workshop and that is the only way how I know to do that :)

Looking into that for the first time, it confused me too. Reading now again through the original rex setup, it used the file in the parent directory, which is more obvious so I will move it there.

Reply:
OK, wow, I'll update the README =)

Reply:
On a second thought, a wizard would allow to drop the section altogether. Feel free to either do it yourself or leave it to me.

Reply:
Will do :)

Reply:
Well, I will probably just update the documentation that the wizard takes care of this. I like the description of the technical details, rather than seeing a black box. See the current HEAD.